### ISSUE TITLE
Solr test failures

### ISSUE THREAD

**Issue by jywarren** [publiclab/plots2#992]
On running tests, I see the following, though that doesn't cause a Travis failure:

```

===============================================================================
Failure:
  <"Notes or Wiki updates"> expected but was
  <nil>.
test_should_create_new_search(SearchesControllerTest)
test/solr/searches_controller_test.rb:48:in `block in <class:SearchesControllerTest>'
===============================================================================
.....F
===============================================================================
Failure:
  <"User Profiles"> expected but was
  <"Notes or Wiki updates">.
test_should_update_advanced_search(SearchesControllerTest)
test/solr/searches_controller_test.rb:65:in `block in <class:SearchesControllerTest>'
===============================================================================
```

See [full build here](https://github.com/publiclab/plots2/commit/50a33054d80c2f50d1377a0a8b53b005bf454aa7#diff-66694174e90422724c8c508d12a0f225L89).

The first error is on this line: https://github.com/publiclab/plots2/blob/master/test/solr/searches_controller_test.rb#L48

By looking at the corresponding controller code, I don't see a change to `assigns(:search).main_type`, but that refers to `@search` which is a method of SearchRecord: https://github.com/publiclab/plots2/blob/master/app/models/search_record.rb

But it's just a db column of that model, so then I did `grep -r main_type app` and saw that it's supposed to be set in the form template: https://github.com/publiclab/plots2/blob/master/app/views/searches/_form.html.erb#L8-L9

Something similar is happening in the second error, here: https://github.com/publiclab/plots2/blob/master/test/solr/searches_controller_test.rb#L65

It's not clear to me why this isn't working, or when it happened, but by looking at the "blame" view we can see what recent changes there are and who did them:

https://github.com/publiclab/plots2/blame/master/app/controllers/searches_controller.rb

I touched the form template about 14 days ago... https://github.com/publiclab/plots2/commit/50a33054d80c2f50d1377a0a8b53b005bf454aa7

Hmm. 

****

Looking further, those should be set on these lines: https://github.com/publiclab/plots2/blob/master/test/solr/searches_controller_test.rb#L42 in the tests. I'm going to hunt through [the old builds](https://travis-ci.org/publiclab/plots2/builds/) to find when this started. 

****

OK -- this commit did it: https://github.com/publiclab/plots2/commit/50a33054d80c2f50d1377a0a8b53b005bf454aa7

Probably the changes here, no? https://github.com/publiclab/plots2/commit/50a33054d80c2f50d1377a0a8b53b005bf454aa7#diff-66694174e90422724c8c508d12a0f225L89

```diff
 -      params.require(:search).permit(:key_words, :main_type, :note_type,
 -                                     :min_date, :max_date, :created_by, :language)
 +      params.permit( :key_words,
 +                     :main_type,
 +                     :note_type,
 +                     :min_date,
 +                     :max_date,
 +                     :created_by,
 +                     :language )
```

@david-days - any ideas on fixing this test?

**Comment by jywarren** [comment]
https://travis-ci.org/publiclab/plots2/builds/176433885

**Comment by jywarren** [comment]
@icarito - sorry, i know you're working on Solr things, but I noticed that this set of tests failed, and I can fix them -- but they didn't actually trigger a failure of TravisCI -- is there something we need to do to ensure they cause a failure?

**Comment by jywarren** [comment]
I'm thinking that since I removed `params.require(:search)` I should've added it to the `permit` list. Or removed it. Attempting fix in #997.
